codec-classes-quic/src/main/java/io/netty/handler/codec/quic/QuicheQuicConnection.java

```diff
@@ -60,6 +60,9 @@ final class QuicheQuicConnection {
     private long connection;
 
     QuicheQuicConnection(long connection, long ssl, QuicheQuicSslEngine engine, ReferenceCounted refCnt) {
+        if (connection == -1) {
+            throw new IllegalArgumentException("Unset connection");
+        }
```

Comment:
This bit isn't strictly necessary, but figured it can't hurt

Reply:
maybe just make it an assert ?